MAINT: Start applying ruff/flake8-comprehensions rules (C4)

DimitriPapadopoulos opened the pull request:
In preparation for https://github.com/numpy/numpy/issues/24994.

DimitriPapadopoulos committed:
MAINT: Apply ruff/flake8-comprehensions rule C410

C410 Unnecessary `list` literal passed to `list()`
     (remove the outer call to `list()`)

DimitriPapadopoulos committed:
MAINT: Apply ruff/flake8-comprehensions rule C414

C414 Unnecessary `list` call within `sorted()`

DimitriPapadopoulos committed:
MAINT: Apply ruff/flake8-comprehensions rule C416

C416 Unnecessary `list` comprehension (rewrite using `list()`)

DimitriPapadopoulos committed:
MAINT: Apply ruff/flake8-comprehensions rule C417

C417 Unnecessary `map` usage (rewrite using a generator expression)

DimitriPapadopoulos committed:
MAINT: Apply ruff/flake8-comprehensions rule C419

C419 Unnecessary list comprehension

mhvk commented:
This also has linting errors! (So I didn't review yet)

DimitriPapadopoulos commented:
The linting errors are not caused by existing code, not by my changes. I would like to add `noqa: E231` in this specific case.  Thoughts?

mhvk commented:
Probably best to just fix them - best not to merge PRs which do not pass the tests...

DimitriPapadopoulos committed:
MAINT: Apply ruff/Pycodestyle rule E231

E231 Missing whitespace after ','

Apply to a single file to please the current linter.

DimitriPapadopoulos commented:
Should be fixed by running ` ruff check --select E231` on this specific file (requires `--preview` by the way): 
```console
$ ruff check --preview --select E231 numpy/_core/tests/test_nditer.py --fix
Found 33 errors (33 fixed, 0 remaining).
$ 
```

mhvk commented on numpy/_core/tests/test_cython.py:
```diff
@@ -217,10 +217,8 @@ def test_multiiter_fields(install_temp, arrays):
-        [
-            x.base is y.base
+        x.base is y.base
             for x, y in zip(bcast.iters, checks.get_multiiter_iters(bcast))
```
Am surprised that this weird indentation passes the linter...

mhvk commented on numpy/_core/tests/test_cython.py:
```diff
@@ -280,10 +278,8 @@ def test_npyiter_api(install_temp):
-        [
-            np.allclose(x, y)
+        np.allclose(x, y)
             for x, y in zip(checks.get_npyiter_itviews(it), it.itviews)
```
And another one.

mhvk reviewed (commented):
Hmm, weird that that was the problem, especially as there seem to be worse problems (at least visually).

charris commented on numpy/ma/tests/test_core.py:
```diff
@@ -3789,19 +3789,19 @@ def test_toflex(self):
+        data = array(list(zip(np.arange(10),
                                                      'ABCDEFGHIJKLM',
-                                                     np.random.rand(10))],
+                                                     np.random.rand(10))),
```
This could use some cleanup.

charris commented on numpy/ma/tests/test_core.py:
```diff
@@ -3789,19 +3789,19 @@ def test_toflex(self):
+        data = array(list(zip(np.arange(10),
                                                        np.random.rand(10),
-                                                       np.random.rand(10))],
+                                                       np.random.rand(10))),
```
This could use some cleanup.

DimitriPapadopoulos replied:
You mean indentation cleanup, don't you?

charris reviewed (changes requested):
Couple of ugly spots, otherwise looks OK. A few tricky bits, but mostly straightforward.

charris commented:
Thanks @DimitriPapadopoulos .
